File: gym-management-api/src/main/java/com/gym/api/gym_management_api/domain/model/Vinculo.java
package com.gym.api.gym_management_api.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Vinculo {

  private Long id;
  private Long usuarioId;
  private Long academiaId;
  private Long personalId;
  private Boolean ativo;
  private LocalDateTime dataVinculo;
  private LocalDateTime dataAtualizacao;

  public void ativar() {
    this.ativo = true;
  }

  public void inativar() {
    this.ativo = false;
  }
}

File: client/src/types/vinculo.ts
import type { UsuarioResponse } from "./auth";

export interface VinculoResponse {
  id: number;
  usuario: UsuarioResponse;
  academiaId: number;
  personal: UsuarioResponse | null;
  ativo: boolean;
  dataVinculo: string;
}

export interface CreateVinculoRequest {
  usuarioId: number;
  academiaId: number;
  personalId?: number;
}
